feat: Select darkmode in app.cal.com

swh00tw opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Add a section in the appearance page to switch dark/light mode for the Main page(not the booking page).

Solved #11997

<!-- Please provide a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

https://github.com/calcom/cal.com/assets/71842426/6c165617-283e-4aee-9845-8b3e660985d2


## Requirement/Documentation

<!-- Please provide all documents that are important to understand the reason of that PR. -->

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

<!-- Please delete bullets that are not relevant. -->

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

1. Go to `localhost:3000`
2. Go to your setting page's appearance tab
3.  Switch the "App theme", and you will see the color change of the UI. (Cal.com's system default is dark mode)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

<!-- Remove bullet points below that don't apply to you -->

## Note

Currently, for the wording here, it's only in English for now (I've asked a person from Cal.com in the OSS hackathon, and he said using English for now is good). We need to create another PR to support all other languages.

![Screenshot 2023-12-02 at 12 12 24 PM](https://github.com/calcom/cal.com/assets/71842426/a1ea63bf-bc98-4b4e-9121-97113e698351)

swh00tw committed:
feat(db): add new col: user appTheme

swh00tw committed:
feat(event-types page): apply appTheme color

swh00tw committed:
add appTheme for trpc

swh00tw committed:
feat: mutate main page theme

swh00tw committed:
feat: add english translation for app theme update section

swh00tw committed:
modify keys in common (en)

swh00tw committed:
add woring

swh00tw committed:
Revert yarn.lock to its state before unintended changes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=12633) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=12633) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=12633) it.</sub>

Udit-takkar commented on apps/web/pages/settings/my-account/appearance.tsx:
```diff
@@ -137,6 +151,56 @@ const AppearanceView = ({
       <Meta title={t("appearance")} description={t("appearance_description")} borderInShellHeader={false} />
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: currently it's hard code now, we should support other languages in other PR */}
```
you have to just add translation for english language in apps/web/public/static/locales/en/common.json. we'll add translations for other languages later.

Udit-takkar replied:
For example:- If you have to use text "Reason for reschedule".
add a key value pair in en/common.json "reason_for_reschedule" :  "Reason for reschedule". 

and use it here t("reason_for_reschedule") like you can see other parts of code

swh00tw replied:
Gotcha, I think I did that. I added a new key called "app_theme". However, the vscode extension kept saying something like "ERROR: "app_theme" is not found in source with language tag 'en'". And the website isn't reflecting the new wording as well. Still trying to figure out why.

<img width="671" alt="Screenshot 2023-12-02 at 12 43 01 PM" src="https://github.com/calcom/cal.com/assets/71842426/2c259bb9-bae1-46ec-9321-8850c790e2f5">

<img width="1032" alt="Screenshot 2023-12-02 at 12 44 30 PM" src="https://github.com/calcom/cal.com/assets/71842426/73ebd4f3-460d-439f-8461-011c1087d134">

swh00tw replied:
After I `yarn dx`. The website has reflected the changes. Nice. But the vscode extension warning is still there

Udit-takkar replied:
@swh00tw no issue. this happens sometimes. just make sure you have added all the keys to common.json that you are using even though you see this error in extension. it gets fixed after some time

Udit-takkar commented:
@swh00tw there are some type error and make sure to sign CLA https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236

swh00tw commented:
> @swh00tw there are some type error and make sure to sign CLA [#12633 (comment)](https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236)

Okay! I am working on getting rid of that now.

swh00tw committed:
feat(i18n): add back the i18n string (for dark/light mode switching)

swh00tw committed:
refactor: refactor ThemeLabel

swh00tw committed:
fix: add new user field "appTheme" to test files

swh00tw commented:
> @swh00tw there are some type errors and make sure to sign CLA [#12633 (comment)](https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236)

Solved the typing error. Notify me if there is any problem :)

swh00tw committed:
chore: modify TODO comment

swh00tw committed:
chore: update common.json (en) for "theme"

since we had appTheme now, I think it's better to update the wording here to avoid confusion

swh00tw committed:
fix: update button's data-testid to fix e2e test error

Udit-takkar commented on apps/web/pages/settings/my-account/appearance.tsx:
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
       <Meta title={t("appearance")} description={t("appearance_description")} borderInShellHeader={false} />
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: we should support other languages in other PR */}
```
NIT: remove this

swh00tw replied:
Removed

Udit-takkar commented on apps/web/pages/settings/my-account/appearance.tsx:
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
+        form={userAppThemeFormMethods}
+        handleSubmit={(values) => {
+          mutation.mutate({
+            // Radio values don't support null as values, therefore we convert an empty string
+            // back to null here.
```
NIT: you can remove this comment

swh00tw replied:
Removed

Udit-takkar commented on apps/web/pages/settings/my-account/appearance.tsx:
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: we should support other languages in other PR */}
+          <p className="text-default text-base font-semibold">{t("app_theme")}</p>
+          <p className="text-default">{t("app_theme_applies_note")}</p>
```
we should use the text mentioned in the issue comment by peer

https://github.com/calcom/cal.com/issues/11997#issuecomment-1828153608

"Dashboard theme"

swh00tw replied:
Got it! I've modified the text. I should read the issue more carefully next time. Thanks!

Udit-takkar reviewed (commented):
@swh00tw Nice attempt but there are some issues

When I tried to change the app theme then https://www.loom.com/share/98f941133c5a46fd9c7b86f6ce10ca97?sid=b178f310-0963-48f9-856d-414996203b53

there were some issues with colors and i had to reload to make sure correct colors are loaded

swh00tw committed:
chore: remove comment

Udit-takkar commented:
Not only the toast component but even buttons, switches .. etc.

swh00tw commented:
I see. I can reproduce this error. Do you have any idea why this happened or any suggested way to resolve this?
I am suspecting that maybe `useTheme` miss something so that the styles of the components aren't not changed

Udit-takkar commented:
@sean-brydon might know about this

swh00tw committed:
fix: fix brand-color-not-apply bug

swh00tw committed:
solve type error

swh00tw committed:
fix: fix type error

PeerRich committed:
Merge branch 'main' into main

PeerRich committed:
Merge branch 'main' into main

keithwillcode committed:
Merge branch 'main' into main

PeerRich committed:
Merge branch 'main' into main

keithwillcode committed:
Merge branch 'main' into main

PeerRich committed:
Merge branch 'main' into main

Udit-takkar committed:
Merge branch 'main' into main

Udit-takkar reviewed (commented):
If i select 'Light mode' in Dashboard Theme and 'Dark Mode' in Booking Page Theme and go the booking page then i see the booking page in dark mode but the dashboard also changes to dark mode. can we fix this?

https://github.com/calcom/cal.com/assets/53316345/334e8722-b0c9-434b-b1c2-375e6b92f5a2

Udit-takkar commented:
cc: @sean-brydon for any feedback

PeerRich committed:
Merge branch 'main' into main

PeerRich commented:
@swh00tw can you take a look at [this](https://github.com/calcom/cal.com/pull/12633#pullrequestreview-1838258444)?

sean-brydon commented on apps/web/pages/event-types/index.tsx:
```diff
@@ -930,6 +931,7 @@ const EventTypesPage = () => {
   const orgBranding = useOrgBranding();
   const routerQuery = useRouterQuery();
   const filters = getTeamsFiltersFromQuery(routerQuery);
+  useTheme(user?.appTheme);
```
Do we only want to do this in `event-types/index` this should probably be handled in `Shell:356` - This is where we also handle settings brandColours for a user

sean-brydon reviewed (changes requested):
Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned. 

If these changes are not addressed we will take over this PR in a few days.

swh00tw commented:
> Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned.
> 
> If these changes are not addressed we will take over this PR in a few days.

Hey guys, really sorry I missed the previous comments. I am kinda busy these days and have little time working on this. I think the best option for this project is to let you guys take over this PR.

sean-brydon commented:
> > Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned.
> > If these changes are not addressed we will take over this PR in a few days.
> 
> Hey guys, really sorry I missed the previous comments. I am kinda busy these days and have little time working on this. I think the best option for this project is to let you guys take over this PR.

Awesome - thanks for your response. I will take over this PR sometime next week when i get time. Thanks for your contribution.

keithwillcode commented:
Closing in favor of https://github.com/calcom/cal.com/pull/13779
